### ISSUE TITLE
Batch Update in Cross domain access error 

### ISSUE THREAD

**Issue by ahsanranjha** [SharePoint/PnP-JS-Core#797]
![image](https://user-images.githubusercontent.com/9005752/39512565-7d6519ba-4e02-11e8-8c6f-e7cb0ddee60d.png)

  let web = $pnp.sp.crossDomainWeb(JobList.SPAppWebUrl, JobList.SPHostUrl);

                            let list = $pnp.sp.crossDomainWeb(JobList.SPAppWebUrl, JobList.SPHostUrl).lists.getByTitle("Candidates");

                            list.getListItemEntityTypeFullName().then(entityTypeFullName => {

                                let batch = web.createBatch();

                                // note requirement of "*" eTag param - or use a specific eTag value as needed
                                list.items.getById(TasksSPItemCollection[0].data.Candidate_LookupId).inBatch(batch).update({ Title: "Batch 6" }, "*", entityTypeFullName).then(b => {
                                    console.log(b);
                                });

                                list.items.getById(TasksSPItemCollection[1].data.Candidate_LookupId).inBatch(batch).update({ Title: "Batch 7" }, "*", entityTypeFullName).then(b => {
                                    console.log(b);
                                });

                                batch.execute().then(d => console.log("Done"));
                            });

**Source Code - PnP**

let list = pnp.sp.web.lists.getByTitle("rapidadd");

list.getListItemEntityTypeFullName().then(entityTypeFullName => {

    let batch = pnp.sp.web.createBatch();

    list.items.inBatch(batch).add({ Title: "Batch 6" }, entityTypeFullName).then(b => {
        console.log(b);
    });

    list.items.inBatch(batch).add({ Title: "Batch 7" }, entityTypeFullName).then(b => {
        console.log(b);
    });

    batch.execute().then(d => console.log("Done"));
});

**Comment by koltyakov** [comment]
The same answer as from the sibling thread:

I created a SPHostedApp and checked:

(async () => {

	const web = pnp.sp.crossDomainWeb(
            GetUrlKeyValue('SPAppWebUrl'),
            GetUrlKeyValue('SPHostUrl')
        );

	const batch = pnp.sp.web.createBatch(); // worked
	// const batch = pnp.sp.createBatch(); // worked
        // const batch = web.createBatch(); // didn't work,
            // my assumption that a batch should be created from `crossDomainWeb` web object was faulty

	// code is omitted here ...

	await batch.execute();
	console.log('Done');

})().catch(console.log);

In other words, everything fine with batches in SP-Hosted Apps, requests should be created using `pnp.sp.crossDomainWeb` object, but batch object should be created with `pnp.sp.createBatch()` or `pnp.sp.web.createBatch()`, then as usual.

**Comment by patrick-rodgers** [comment]
Thank you for your interest in the sp-pnp-js library. We wanted to mention that this library is being depricated in July, 2018 in favor of the [new scoped pnpjs libraries](https://github.com/pnp/pnpjs). You should begin transitioning your existing projects when possible, or start new projects with the new libraries. Please see the [transition guide](https://pnp.github.io/pnpjs/transition-guide.html) for more details on migrating and be sure to let us know if you have any questions. Thanks!

**Comment by ahsanranjha** [comment]
July,2018 is future date. I assume as July 2017, is there any possibility to update sp-pnp-js library with rest api upgraded with followings, It is always required to create multiple items inside a folder. 

/_api/SP.AppContextSite(@target)/web/lists/GetByTitle('Sample')/AddValidateUpdateItemUsingPath() 

???

**Comment by koltyakov** [comment]
@ahsanranjha, please, don't bombard the same questions in separate threads. One is enough and easier for us to respond and track. Thanks for understanding.
